org.eclipse.lsp4e/src/org/eclipse/lsp4e/ui/LanguageServersView.java

```diff
@@ -224,14 +258,81 @@ public void dispose() {
 		super.dispose();
 	}
 
+	private void initContextMenu() {
+		// Avoiding the use of MenuManager, table.setMenu(contextMenu), and
+		// SWT.MenuDetect because the table's SWT.FULL_SELECTION interferes with
+		// right-click detection, causing inconsistent behavior and preventing the
+		// context menu from being displayed reliably.
+		final Table table = viewer.getTable();
+		final var contextMenu = new Menu(table);
+		final var copyValueMenuItem = new MenuItem(contextMenu, SWT.NONE);
+		copyValueMenuItem.setText("Copy Cell Value"); //$NON-NLS-1$
+		copyValueMenuItem.addListener(SWT.Selection, event -> copyCurrentCellToClipboard());
+		table.addListener(SWT.MouseDown, event -> {
+			if (event.button == 3 /* Right-click? */
```

Comment:
very surprising that there is no better way to find if it is a right-click that hardcoding the number 3, but I could not find anything better in SWT; the documentation only says the button can be 1, 2, 3.

Reply:
Yes, I was surprised by this as well. Ideally, I would use SWT.MenuDetect as the listener event, which would make this check unnecessary. However, for some reason, SWT.MenuDetect becomes unreliable when SWT.FULL_SELECTION is enabled on the table. Then it only triggers intermittently after several right mouse clicks.